### PR TITLE
Allow setting of first ip for network reservations

### PR DESCRIPTION
**Pull request by treywelsh**
This is a PR that apply some small changes to https://github.com/OpenNebula/terraform-provider-opennebula/pull/185 authored by @danhigham

### CODE REVIEW

**Comment by treywelsh** [comment]
This PR introduce the `reservation_ar_id` attribute.
It's easy to retrieve with an already existing AR but we may want to be able to create a reservation at the same time that we create the AR in the parent network.
In addition, it's not possible to rely on address range order to guess the ID (ARs are stored in a TypeSet. We could store them in a TypeList instead, but keep AR ordering through future AR modification is not so funny).

I don't see any easy and clean way to work with `reservation_ar_id` other than externalize the address range to it's own resource (but it's not the only argument to do this anyway): #279 

I'll remove this issue from 0.5.0 milestone, it will require a bit more of work

**Comment by treywelsh** [comment]
depends on #279 should be scheduled for 1.0 release

**Comment by treywelsh** [comment]
rebased on top of #306 until it's merged to make this PR move forward

File: website/docs/r/virtual_network.html.markdown [hunk 1]
```diff
@@ -20,11 +20,14 @@ a new virtual network is created. When destroyed, that virtual network is remove
 Allocate a new virtual network from the parent virtual network "394":
 
 ```hcl
+<<<<<<< HEAD
```
Comment (treywelsh) [inline website/docs/r/virtual_network.html.markdown]: remove this

File: CHANGELOG.md [hunk 2]
```diff
@@ -142,6 +142,7 @@ FEATURES:
 * resources/opennebula_virtual_machine: Optionally preserve NIC ordering
 * resources/opennebula_virtual_machine: Enable locking
 * resources/opennebula_virtual_network: Enable locking
+* resources/opennebula_virtual_network: Set first IP and address range ID for address reservations
```
Comment (frousselet) [inline CHANGELOG.md:145]: Add attributes `reservation_first_ip` and `reservation_ar_id`

Comment (frousselet) [inline CHANGELOG.md]: Add attributes `reservation_first_ip` and `reservation_ar_id`.

**Review by frousselet — CHANGES_REQUESTED** [review]

**Review by frousselet — APPROVED** [review]

**Pull request merged**